src/main/java/fi/otavanopisto/kuntaapi/server/integrations/vcard/VCardTranslator.java

```diff
@@ -0,0 +1,271 @@
+    if (properties != null) {
+      List<ContactStatus> translateMecmStatuses = new ArrayList<>(properties.size());
+      for (RawProperty property : properties) {
+        translateMecmStatuses.add(translateMecmStatus(property.getValue()));
+      }
+      
+      return translateMecmStatuses;
+    }
+    
+    return Collections.emptyList();
+  }
+
+  private ContactStatus translateMecmStatus(String value) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    objectMapper.registerModule(new JavaTimeModule());
+    
+    try {
+      Status mecmStatus = objectMapper.readValue(value, Status.class);
+      
+      ContactStatus contactStatus = new ContactStatus();
+      contactStatus.setEnd(mecmStatus.getArrival());
+      contactStatus.setStart(mecmStatus.getDeparture());
+      contactStatus.setText(mecmStatus.getReason());
+      
+      return contactStatus;
+    } catch (IOException e) {
+      logger.log(Level.SEVERE, "Failed to parse MECM statuses", e);
+    }
+    
+    return null;
+  }
+
+  private List<ContactPhone> translatePhones(List<Telephone> vCardNumbers) {
+    if (vCardNumbers != null && !vCardNumbers.isEmpty()) {
+      List<ContactPhone> result = new ArrayList<>(vCardNumbers.size());
+      for (Telephone vCardNumber : vCardNumbers) {
+        ContactPhone contactPhone = new ContactPhone();
+        if (StringUtils.isNotBlank(vCardNumber.getText())) {
+          contactPhone.setNumber(vCardNumber.getText());
+          contactPhone.setType(getPhoneType(vCardNumber.getTypes()));
+          result.add(contactPhone);
+        }
+      }
+    }
+    
+    return Collections.emptyList();
+  }
+
+  private List<Address> translateAddresses(List<ezvcard.property.Address> vCardAddresses) {
+    if (vCardAddresses != null && !vCardAddresses.isEmpty()) {
+      List<Address> result = new ArrayList<>(vCardAddresses.size());
+      
+      for (ezvcard.property.Address vCardAddress : vCardAddresses) {
+        Address address = translateAddress(vCardAddress);
+        if (address != null) {
+          result.add(address);
+        }
+      }
+      
+      return result;
+    }
+    
+    return Collections.emptyList();
+  }
+
+  private Address translateAddress(ezvcard.property.Address vCardAddress) {
+    if (vCardAddress == null) {
+      return null;
+    }
+    
+    Address result = new Address();
+    
+    result.setType(getAddressType(vCardAddress.getTypes()));
+    result.setPostOfficeBox(getString(vCardAddress.getPoBox()));
+    result.setPostalCode(getString(vCardAddress.getPostalCode()));
+    result.setPostOffice(getString(vCardAddress.getRegion()));
+    result.setStreetAddress(getLocalizedString(vCardAddress.getStreetAddress()));
+    result.setMunicipality(null);
+    result.setCountry(getString(vCardAddress.getCountry(), VCardConsts.DEFAULT_COUNTRY));
+    result.setQualifier(null);
+
+    return result;
+  }
+
+  private String getAddressType(List<AddressType> types) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getAddressType" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)